extractor/src/test/java/org/schabi/newpipe/downloader/MockDownloader.java

```diff
@@ -0,0 +1,55 @@
+package org.schabi.newpipe.downloader;
+
+import com.google.gson.GsonBuilder;
+
+import org.schabi.newpipe.extractor.downloader.Downloader;
+import org.schabi.newpipe.extractor.downloader.Request;
+import org.schabi.newpipe.extractor.downloader.Response;
+
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.annotation.Nonnull;
+
+class MockDownloader extends Downloader {
+
+    private final String path;
+    private final Map<Request, Response> mocks;
+
+    public MockDownloader(@Nonnull String path) throws IOException {
+        this.path = path;
+        this.mocks = new HashMap<>();
+        File folder = new File(path);
+        for (File file : folder.listFiles()) {
+            final FileReader reader = new FileReader(file);
+            final TestRequestResponse response = new GsonBuilder()
+                    .create()
+                    .fromJson(reader, TestRequestResponse.class);
```

Comment:
I am not sure, but maybe here you should catch errors, so that if we put some files of a different format in the same folder there are no issues (we may want to do that at some point, maybe)

Reply:
I resolved this by prefixing the files and only writing/reading from those.